apps/src/aichat/views/modelCustomization/CompareModelsDialog.tsx

```diff
@@ -1,17 +1,29 @@
 import React, {useState} from 'react';
 
+import {useAppSelector} from '@cdo/apps/util/reduxHooks';
 import AccessibleDialog from '@cdo/apps/templates/AccessibleDialog';
 import {Heading3} from '@cdo/apps/componentLibrary/typography';
 import Button from '@cdo/apps/componentLibrary/button/Button';
 
 import ModelDescriptionPanel from './ModelDescriptionPanel';
 import styles from './compare-models-dialog.module.scss';
+import {ModelDescription} from '../../types';
 
-const CompareModelsDialog: React.FunctionComponent<{onClose: () => void}> = ({
-  onClose,
-}) => {
-  const [chosenModelLeft, setChosenModelLeft] = useState<string>('llama2');
-  const [chosenModelRight, setChosenModelRight] = useState<string>('mistral');
+const CompareModelsDialog: React.FunctionComponent<{
+  onClose: () => void;
+  availableModels: ModelDescription[];
+}> = ({onClose, availableModels}) => {
+  const selectedModelId = useAppSelector(
+    state =>
+      state.aichat.currentAiCustomizations.selectedModel ??
```

Comment:
Just curious about `??` operator which returns the first argument if it is not `null` nor `undefined`.
In `modelCustomization/constants`, the default `selectedModel` is assign `""`. So at some point, could `selectedModel` be assigned `""`?

Reply:
One idea - when we have the levelbuilder side working, we could also make it so that `selectedModel` is always initialized to first available model if not already set by levelbuilder/student, so we don't need this extra fallback here.

Reply:
curious about this too, we have very few usages of this pattern in our code base for some reason

Reply:
I was using it because I'm used to using that operator in c# >.> but I think what I actually want is || here. The differnce being that '' counts as falsy with || but not ??

Reply:
I'm used to || returning a boolean. Javascript is weird.

Reply:
I think changing to || essentially does what Sanchit is suggesting, since if not set by levelbuilder, selectedModel will be '', and availableModels will have been empty and therefore defaulted to all models.